### DISCUSSION TITLE
Custom OpenAPI URL?

### DISCUSSION THREAD

**Discussion by zdcain** [danielmiessler/Fabric#878] [Q&A]
I've just installed the latest Fabric and can't wait to get my feet wet!

I've run into a minor speed bump because I need to use a different OpenAPI URL internal to my environment.  I entered my OPEN_API_KEY during setup but wanted to know if something like OPEN_API_URL was possible.

I want to thank you in advance for your time and consideration.  I'm very excited to increase my augmentation via AI.

**Comment by danhuss** [comment]
It looks like in the Python version the workaround was to manually set the OPENAI_BASE_URL environment variable, but this appears to not work in the Go version. #883

**Comment by bryanmcguire** [comment]
I ended up doing this:
```
diff --git a/vendors/openai/openai.go b/vendors/openai/openai.go
index e9c9755..0e9ea74 100644
--- a/vendors/openai/openai.go
+++ b/vendors/openai/openai.go
@@ -5,6 +5,7 @@ import (
        "errors"
        "fmt"
        "io"
+       "os"

        "github.com/danielmiessler/fabric/common"
        "github.com/samber/lo"
@@ -13,7 +14,11 @@ import (
 )

 func NewClient() (ret *Client) {
-       return NewClientCompatible("OpenAI", "https://api.openai.com/v1", nil)
+    url := os.Getenv("OPENAI_BASE_URL")
+    if url == "" {
+        url = "https://api.openai.com/v1"
+    }
+    return NewClientCompatible("OpenAI", url, nil)
 }

 func NewClientCompatible(vendorName string, defaultBaseUrl string, configureCustom func() error) (ret *Client) {
```
 did a `go build .`, then copied the `fabric` binary into my ~/go/bin directory.
 It's sloppy, but it lets me use chat completion with vllm, mistralai, etc.
